### PR TITLE
Add Ncrack

### PR DESCRIPTION
**Pull request by stephengroat**
@Auxilus 

using #1640, moved downloaded files to `.patch` file. Trying to generate error for #1344 in travis

### CODE REVIEW

**Comment by Auxilus** [comment]
👍  let's see how it goes...

File: packages/ncrack/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+TERMUX_PKG_VERSION=0.5.0
+TERMUX_PKG_SHA256=4dd4abfa7c38edafd0da3a3f2ad76f9fae476e147042def87d027879444bd0b9
+TERMUX_PKG_SRCURL=https://github.com/nmap/ncrack/archive/v${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_BUILD_IN_SRC=yes
+TERMUX_PKG_DEPENDS="openssl, openssh, openssl-dev"
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="ac_cv_func_vsnprintf=no ac_cv_header_unistd=no"
+
+termux_step_pre_configure () {
+	cd opensshlib
+        # sed '/explicit_bzero/d' -i *.c
```
Comment (Auxilus) [inline packages/ncrack/build.sh]: The sed step is required ( afaik from my end... )

**Comment by stephengroat** [comment]
error generated (hidden inside the raw travis log)

**Comment by Auxilus** [comment]
Well, we got whole bunch of errors!

**Comment by Auxilus** [comment]
```
In file included from cleanup.c:23:

[1m/home/builder/.termux-build/_lib/15.2-aarch64-21-v14/bin/../sysroot/usr/include/unistd.h:108:20: [0m[0;1;31merror: [0m[1mexpected ')'[0m

    __attribute__((__sentinel__(1)));

[0;1;32m                   ^

[0m1 error generated.

Makefile:71: recipe for target 'cleanup.o' failed

make[1]: *** [cleanup.o] Error 1

make[1]: Leaving directory '/home/builder/.termux-build/ncrack/src/opensshlib'

Makefile:84: recipe for target 'opensshlib_build' failed

make: *** [opensshlib_build] Error 2

make: *** Waiting for unfinished jobs....

aarch64-linux-android-clang++ -c -I/data/data/com.termux/files/usr/include -Inbase -Insock/include -Iopensshlib -Imodules -DHAVE_CONFIG_H -DNCRACK_NAME=\"Ncrack\" -DNCRACK_URL=\"http://ncrack.org\" -DNCRACK_PLATFORM=\"aarch64-unknown-linux-android\" -DNCRACKDATADIR=\"/data/data/com.termux/files/usr/share/ncrack\" -D_FORTIFY_SOURCE=2  -I.. -I../nsock/include -I../nbase -I../opensshlib -DHAVE_CONFIG_H -Oz -O2 -Wall -fno-strict-aliasing -Wno-attributes ncrack_rdp.cc -o ncrack_rdp.o

aarch64-linux-android-clang++ -c -I/data/data/com.termux/files/usr/include -Inbase -Insock/include -Iopensshlib -Imodules -DHAVE_CONFIG_H -DNCRACK_NAME=\"Ncrack\" -DNCRACK_URL=\"http://ncrack.org\" -DNCRACK_PLATFORM=\"aarch64-unknown-linux-android\" -DNCRACKDATADIR=\"/data/data/com.termux/files/usr/share/ncrack\" -D_FORTIFY_SOURCE=2  -I.. -I../nsock/include -I../nbase -I../opensshlib -DHAVE_CONFIG_H -Oz -O2 -Wall -fno-strict-aliasing -Wno-attributes ncrack_sip.cc -o ncrack_sip.o

[1mncrack_rdp.cc:1982:16: [0m[0;1;35mwarning: [0m[1mcomparison of constant 256 with expression of type 'uint8_t' (aka 'unsigned char') is always true [-Wtautological-constant-out-of-range-compare][0m

    if (length <= sizeof(text))

[0;1;32m        ~~~~~~ ^  ~~~~~~~~~~~~

[0m[1mncrack_rdp.cc:1985:16: [0m[0;1;35mwarning: [0m[1mcomparison of constant 256 with expression of type 'uint8_t' (aka 'unsigned char') is always false [-Wtautological-constant-out-of-range-compare][0m

    if (length > sizeof(text)) {

[0;1;32m        ~~~~~~ ^ ~~~~~~~~~~~~

[0m[1mncrack_rdp.cc:2497:10: [0m[0;1;35mwarning: [0m[1mlogical not is only applied to the left hand side of this bitwise operator [-Wlogical-not-parentheses][0m

    if ((!flags & RDP_ORDER_STANDARD)) {

[0;1;32m         ^      ~

[0m[1mncrack_rdp.cc:2497:10: [0m[0;1;30mnote: [0madd parentheses after the '!' to evaluate the bitwise operator first[0m

    if ((!flags & RDP_ORDER_STANDARD)) {

[0;1;32m         ^

[0m[0;32m          (                         )

[0m[1mncrack_rdp.cc:2497:10: [0m[0;1;30mnote: [0madd parentheses around left hand side expression to silence this warning[0m

    if ((!flags & RDP_ORDER_STANDARD)) {

[0;1;32m         ^

[0m[0;32m         (     )

[0maarch64-linux-android-clang++ -c -I/data/data/com.termux/files/usr/include -Inbase -Insock/include -Iopensshlib -Imodules -DHAVE_CONFIG_H -DNCRACK_NAME=\"Ncrack\" -DNCRACK_URL=\"http://ncrack.org\" -DNCRACK_PLATFORM=\"aarch64-unknown-linux-android\" -DNCRACKDATADIR=\"/data/data/com.termux/files/usr/share/ncrack\" -D_FORTIFY_SOURCE=2  -I.. -I../nsock/include -I../nbase -I../opensshlib -DHAVE_CONFIG_H -Oz -O2 -Wall -fno-strict-aliasing -Wno-attributes ncrack_redis.cc -o ncrack_redis.o

aarch64-linux-android-clang++ -c -I/data/data/com.termux/files/usr/include -Inbase -Insock/include -Iopensshlib -Imodules -DHAVE_CONFIG_H -DNCRACK_NAME=\"Ncrack\" -DNCRACK_URL=\"http://ncrack.org\" -DNCRACK_PLATFORM=\"aarch64-unknown-linux-android\" -DNCRACKDATADIR=\"/data/data/com.termux/files/usr/share/ncrack\" -D_FORTIFY_SOURCE=2  -I.. -I../nsock/include -I../nbase -I../opensshlib -DHAVE_CONFIG_H -Oz -O2 -Wall -fno-strict-aliasing -Wno-attributes ncrack_psql.cc -o ncrack_psql.o

3 warnings generated.

aarch64-linux-android-clang++ -c -I/data/data/com.termux/files/usr/include -Inbase -Insock/include -Iopensshlib -Imodules -DHAVE_CONFIG_H -DNCRACK_NAME=\"Ncrack\" -DNCRACK_URL=\"http://ncrack.org\" -DNCRACK_PLATFORM=\"aarch64-unknown-linux-android\" -DNCRACKDATADIR=\"/data/data/com.termux/files/usr/share/ncrack\" -D_FORTIFY_SOURCE=2  -I.. -I../nsock/include -I../nbase -I../opensshlib -DHAVE_CONFIG_H -Oz -O2 -Wall -fno-strict-aliasing -Wno-attributes ncrack_ftp.cc -o ncrack_ftp.o

[1mncrack_psql.cc:289:7: [0m[0;1;35mwarning: [0m[1marray index 35 is past the end of the array (which contains 35 elements) [-Warray-bounds][0m

      response_hex[MD5_DIGEST_LENGTH * 2 + 3] = '\0';

[0;1;32m      ^            ~~~~~~~~~~~~~~~~~~~~~~~~~

[0m[1mncrack_psql.cc:255:3: [0m[0;1;30mnote: [0marray 'response_hex' declared here[0m

  char response_hex[MD5_DIGEST_LENGTH *2 + 3];

[0;1;32m  ^

[0m1 warning generated.
```

**Pull request closed without merging**